internals/secrethub/service_ls.go

```diff
@@ -67,3 +93,46 @@ func (cmd *ServiceLsCommand) Run() error {
 
 	return nil
 }
+
+type serviceTable interface {
+	header() []string
+	row(service *api.Service) []string
+}
+
+type baseServiceTable struct{}
+
+func (sw baseServiceTable) header() []string {
+	return []string{"ID", "DESCRIPTION"}
+}
+
+func (sw baseServiceTable) row(service *api.Service) []string {
+	return []string{service.ServiceID, service.Description}
+}
+
+type keyServiceTable struct {
+	baseServiceTable
+}
+
+func (sw keyServiceTable) header() []string {
+	return append(sw.baseServiceTable.header(), "TYPE")
+}
+
+func (sw keyServiceTable) row(service *api.Service) []string {
+	return append(sw.baseServiceTable.row(service), string(service.Credential.Type))
+}
+
+type awsServiceTable struct {
+	baseServiceTable
+}
+
+func (sw awsServiceTable) header() []string {
+	return append(sw.baseServiceTable.header(), "ROLE", "KMS KEY")
+}
+
+func (sw awsServiceTable) row(service *api.Service) []string {
+	return append(sw.baseServiceTable.row(service), service.Credential.Metadata[api.CredentialMetadataAWSRole], service.Credential.Metadata[api.CredentialMetadataAWSKMSKey])
```

Comment:
`api.CredentialMetadataAWSRole` and `api.CredentialMetadataAWSKMSKey` are pretty lengthy names, starting to wonder if simply using a `aws_role` string here would be more self-documenting...

Reply:
These keys are part of the implicit contract we have between the client and CLI. By using constants to refer to them, we make this contract a bit more explicit and less error-prone, as we ensure the same key is used in all places.

Reply:
Alternatively, we could add a function in the client that translates an `api.Service` to an aws service type, which explicitly has these fields.

```Go
type AWSService struct {
	*api.Service
	Role   string
	KMSkey string
}

func ToAWSService(service *api.Service) (AWSService, error) {
	if service.Credential.Type != api.CredentialTypeAWSSTS {
		// err
	}

	role, ok := service.Credential.Metadata[api.CredentialMetadataAWSRole]
	if !ok {
		// err
	}

	kmsKey, ok := service.Credential.Metadata[api.CredentialMetadataAWSKMSKey]
	if !ok {
		// err
	}

	return AWSService{
		Service: service,
		Role:    role,
		KMSkey:  kmsKey,
	}, nil
}
```

Reply:
Alright, let's just stick with the constants for now.